b2sdk/v1/sync.py

```diff
@@ -0,0 +1,126 @@
+        :param exclude_all_symlinks: if True, exclude all symlinks
+        :type exclude_all_symlinks: bool
+        :param exclude_modified_before: optionally exclude file versions modified before (in millis)
+        :type exclude_modified_before: int, optional
+        :param exclude_modified_after: optionally exclude file versions modified after (in millis)
+        :type exclude_modified_after: int, optional
+        """
+        if include_file_regexes and not exclude_file_regexes:
+            raise v2_exception.InvalidArgument(
+                'include_file_regexes',
+                'cannot be used without exclude_file_regexes at the same time'
+            )
+
+        self._exclude_dir_set = v2.RegexSet(exclude_dir_regexes)
+        self._exclude_file_because_of_dir_set = v2.RegexSet(
+            map(v2.convert_dir_regex_to_dir_prefix_regex, exclude_dir_regexes)
+        )
+        self._exclude_file_set = v2.RegexSet(exclude_file_regexes)
+        self._include_file_set = v2.RegexSet(include_file_regexes)
+        self.exclude_all_symlinks = exclude_all_symlinks
+        self._include_mod_time_range = v2.IntegerRange(
+            exclude_modified_before, exclude_modified_after
+        )
+
+
+DEFAULT_SCAN_MANAGER = ScanPoliciesManager()
+
+
+def zip_folders(folder_a, folder_b, reporter, policies_manager=DEFAULT_SCAN_MANAGER):
+    return v2.zip_folders(folder_a, folder_b, reporter, policies_manager=policies_manager)
+
+
+class Synchronizer(v2.Synchronizer):
+    def __init__(
+        self,
+        max_workers,
+        policies_manager=DEFAULT_SCAN_MANAGER,
+        dry_run=False,
+        allow_empty_source=False,
+        newer_file_mode=v2.NewerFileSyncMode.RAISE_ERROR,
+        keep_days_or_delete=v2.KeepOrDeleteMode.NO_DELETE,
+        compare_version_mode=v2.CompareVersionMode.MODTIME,
+        compare_threshold=None,
+        keep_days=None,
+    ):
+        super().__init__(
+            max_workers, policies_manager, dry_run, allow_empty_source, newer_file_mode,
+            keep_days_or_delete, compare_version_mode, compare_threshold, keep_days
+        )
+
+    def make_folder_sync_actions(
+        self,
+        source_folder,
+        dest_folder,
+        now_millis,
+        reporter,
+        policies_manager=DEFAULT_SCAN_MANAGER,
+        encryption_settings_provider=v2.SERVER_DEFAULT_SYNC_ENCRYPTION_SETTINGS_PROVIDER,
+    ):
+        return super().make_folder_sync_actions(
+            source_folder, dest_folder, now_millis, reporter, policies_manager,
+            encryption_settings_provider
+        )
+
+
+class AbstractFolder(v2.AbstractFolder):
+    @abstractmethod
+    def all_files(self, reporter, policies_manager=DEFAULT_SCAN_MANAGER):
+        pass
+
+
+class LocalFolder(v2.LocalFolder, AbstractFolder):
+    pass
+
+
+class B2Folder(v2.B2Folder, AbstractFolder):
+    pass
+
+
+def parse_sync_folder(folder_name, api, local_folder_class=LocalFolder, b2_folder_class=B2Folder):
```

Comment:
If possible, could you remove the possibility to provide `local_folder_class` and `b2_folder_class` from `v1.parse_sync_folders`? I am working on something that touches this function as well, and having it return any user provided class would break my code.

Reply:
I would have to copy-past both two methods to `v1.sync` as they have hardcoded classes inside. I can do it

Reply:
No i don't mean changing anything in the actual sdk codebase, just in this v1 delegate:

```
def parse_sync_folder(folder_name, api):
    """
    Take either a local path, or a B2 path, and returns a Folder
    object for it.
    :param folder_name: a name of the folder, either local or remote
    :type folder_name: str
    :param api: an API object
    :type api: b2sdk.api.B2Api
    :type api: :class:`~b2sdk.v1.B2Api`
    """
    if folder_name.startswith('b2://'):
        return _parse_bucket_and_folder(folder_name[5:], api, B2Folder)
    elif folder_name.startswith('b2:') and folder_name[3].isalnum():
        return _parse_bucket_and_folder(folder_name[3:], api, B2Folder)
    else:
        return LocalFolder(folder_name)
```

Reply:
Ahh, that's easy. Consider it done

Reply:
It's even better that way